e2e-tests/test/debug-apis.test.ts

```diff
@@ -72,4 +73,45 @@ describe("debug namespace", function () {
     const [output_number] = primary.interface.decodeFunctionResult("calculate", output);
     expect(output_number.toNumber()).to.equal(12);
   });
+
+  it("Should return null if txn hash is unknown", async function () {
+    const result = await provider.send("debug_traceTransaction", [
+      "0xd3a94ff697a573cb174ecce05126e952ecea6dee051526a3e389747ff86b0d99",
+    ]);
+    expect(result).to.equal(null);
+  });
+
+  it("Should trace prior transactions", async function () {
+    const wallet = new Wallet(RichAccounts[0].PrivateKey);
+    const deployer = new Deployer(hre, wallet);
+
+    const greeter = await deployContract(deployer, "Greeter", ["Hi"]);
+
+    const txReceipt = await greeter.setGreeting("Luke Skywalker");
+    const trace = await provider.send("debug_traceTransaction", [txReceipt.hash]);
+
+    // call should be successful
+    expect(trace.error).to.equal(null);
+    expect(trace.calls.length).to.equal(1);
```

Comment:
I would've expected more than 1 trace call... I would make sure that when you use `call_traces.clone()`, perhaps we should be using `call_traces[0]` instead

Reply:
This is the top-level call to `BOOTLOADER_CONTRACT`, the `calls` field of this will contain subcalls, and so on, eventually we see calls to the contracts. Or you expect something different?

Reply:
AH, indeed... I was thinking of how we implemented the console pretty print of the subcalls.